src/pulselistener/pulselistener/listener.py

```diff
@@ -280,28 +247,17 @@ def __init__(self,
         )
         self.pulse.register(self.bus)
 
-        # Phabricator publication
-        self.phabricator = PhabricatorCodeReview(
-            api=phabricator_api,
-            publish=taskcluster.secrets['PHABRICATOR'].get('publish', False),
-        )
-        self.bus.add_queue(QUEUE_PHABRICATOR_RESULTS)
-
     def run(self):
 
-        # Build hooks for each conf
-        hooks = [
-            self.build_hook(conf)
-            for conf in self.hooks_configuration
-        ]
-        if not hooks:
-            raise Exception('No hooks created')
-
+        # Run client applications
         consumers = [
-            hook.run()
-            for hook in hooks
+            client.run()
+            for client in self.clients
         ]
 
+        # Add build details loading step
+        consumers.append(self.phabricator.load_builds(QUEUE_WEB_BUILDS, QUEUE_CODE_REVIEW))
```

Comment:
Could we move this directly into PhabricatorCodeReview? Maybe after we get rid of pulselistener